modules/navigation/2d/nav_mesh_generator_2d.cpp

```diff
@@ -87,8 +87,8 @@ void NavMeshGenerator2D::sync() {
 		return;
 	}
 
-	baking_navmesh_mutex.lock();
-	generator_task_mutex.lock();
+	MutexLock baking_navmesh_lock(baking_navmesh_mutex);
+	MutexLock generator_task_lock(generator_task_mutex);
```

Comment:
While it doesn't seem to cause any problems I feel that locking two mutexes like this is a recipe for starvation, so would be good to look into that down the line